File: src/main/java/com/elchologamer/userlogin/database/sql/SQLite.java
package com.elchologamer.userlogin.database.sql;

import java.io.File;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class SQLite extends SQLDatabase {

    public SQLite() {
        super("sqlite", "org.sqlite.JDBC");
    }

    @Override
    protected Connection getConnection() throws SQLException {
        String path = getPlugin().getConfig().getString(
                "database.sqlite.database",
                "C:/sqlite/db/userlogin.db"
        );

        File folder = new File(path).getParentFile();
        folder.mkdirs();

        return DriverManager.getConnection("jdbc:sqlite:" + path);
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/database/sql/SQLite.kt
package com.elchologamer.userlogin.database.sql

import com.elchologamer.userlogin.UserLogin.Companion.plugin
import java.io.File
import java.sql.Connection
import java.sql.DriverManager

class SQLite : SQLDatabase("sqlite", "org.sqlite.JDBC") {
    override fun getConnection(): Connection {
        val path = plugin.config.getString("database.sqlite.database", "C:/sqlite/db/userlogin.db")!!
        File(path).parentFile.mkdirs()

        return DriverManager.getConnection("jdbc:sqlite:$path")
    }
}